### ISSUE TITLE
xud crashes when peer crashes after accepting a swap

### ISSUE THREAD

**Issue by erkarl** [ExchangeUnion/xud#1455]
1. Start a swap between A (taker) and B (maker)
2. Crash A after receiving the swap accepted packet
3. Error! B crashes after some timeout.

```
02/04/2020 09:15:27.953 [SWAPS] debug: sending swap accepted packet: {"makerCltvDelta":14664,"rHash":"4b9f8a8cc6901b73650ce51799c7da67c9fa99bf40e477ffde9a4247e36ed85d","quantity":100000} to peer: 02990e0f8d79a7cd8bb9a4652e6494f8c9d9ffcd6a5dd62e4339aac0c53f9185d0
02/04/2020 09:15:29.230 [P2P] info: Peer 02990e0f8d79a7cd8bb9a4652e6494f8c9d9ffcd6a5dd62e4339aac0c53f9185d0 socket closed
02/04/2020 09:15:29.231 [ORDERBOOK] debug: removed all orders for peer 02990e0f8d79a7cd8bb9a4652e6494f8c9d9ffcd6a5dd62e4339aac0c53f9185d0



{
  message: 'node (02990e0f8d79a7cd8bb9a4652e6494f8c9d9ffcd6a5dd62e4339aac0c53f9185d0) is not connected',
  code: '2.2'
}

/home/ar/xud/dist/bootstrap.js:13
        throw err;
        ^
{
  message: 'node (02990e0f8d79a7cd8bb9a4652e6494f8c9d9ffcd6a5dd62e4339aac0c53f9185d0) is not connected',
  code: '2.2'
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! xud@1.0.0-beta.1 start: `node dist/Xud.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the xud@1.0.0-beta.1 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ar/.npm/_logs/2020-04-02T09_15_57_940Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! xud@1.0.0-beta.1 dev: `npm run compile:dev && npm start`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the xud@1.0.0-beta.1 dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ar/.npm/_logs/2020-04-02T09_15_57_956Z-debug.log
```

**Comment by kilrau** [comment]
Just seeing this now... we should probably take care of that. Can you reproduce? @erkarl

**Comment by kilrau** [comment]
Can you reproduce and synce with @sangaman to get the fix in?

**Comment by sangaman** [comment]
I examined the code closely to see how this crash could have happened, I'm pretty sure it will be fixed with #1501 and the `tryGetPeer` method. The `Swaps` class assumes we are connected to the peer and throws an exception if we're not, that exception is handled except when dealing with a timeout (and, unfortunately a swap timeout can easily be caused by a peer becoming disconnected).
